File: soma_de_matrizes.ts
class Matriz {
    private linhas : number;
    private colunas : number;
    private matriz : number[][];

    constructor (linhas : number, colunas : number) {
        this.linhas = linhas;
        this.colunas = colunas;
        this.matriz = this.criarMatrizVazia(linhas, colunas);
    }

    private criarMatrizVazia(linhas : number, colunas : number) : number[][] {
        return new Array (linhas).fill([]).map(() => new Array (colunas).fill(0));
    }

    public somar (outraMatriz : Matriz) : Matriz | null {
        if (this.linhas !== outraMatriz.linhas || this.colunas !== outraMatriz.colunas) {
            console.error("Não é possível somar matrizes de tamanhos diferentes.");
            return null;
        }

    const resultado = new Matriz(this.linhas, this.colunas);

    for (let i = 0; i < this.linhas; i++) {
        for (let j = 0; j < this.colunas; j++) {
            resultado.matriz[i][j] = this.matriz[i][j] + outraMatriz.matriz[i][j];
        }
    }

    return resultado;

   }
}

// Criando objetos da classe Matriz
const matriz_A = new Matriz(3,4);
const matriz_B = new Matriz(3,4);

// Realizando uma operação com os objetos
const resultado_Soma = matriz_A.somar(matriz_B);

// Exibindo o resultado
console.log(resultado_Soma);

File: soma_de_matrizes.java
public class Matriz {
    private int linhas;
    private int colunas;
    private int[][] matriz;

    public Matriz (int linhas, int colunas) {
        this.linhas = linhas;
        this.colunas = colunas;
        this.matriz = new int [linhas][colunas];
    }

    private int[][] criarMatrizVazia(int linhas, int colunas) {
        return new int [linhas][colunas]
    }

    public Matriz somar(Matriz outraMatriz) {
        if (this.linhas !== outraMatriz.linhas || this.colunas !== outraMatriz.colunas) {
            System.err.println("Não é possível somar matrizes de tamanhos diferentes.");
            return null;
        }
    }
    Matriz resultado = new Matriz(this.linhas, this.colunas);

    for (int i = 0; i < this.linhas, i++) {
        for (int j = 0; j < this.colunas, j++) {
            resultado.matriz[i][j] = this.matriz[i][j] + outraMatriz.matriz[i][j];
        }
    }

    return resultado;


    public static void main(String[] args) {
        // Criando objetos da classe Matriz

        Matriz matriz_A = new Matriz(3,4);
        Matriz matriz_B = new Matriz(3,4);

        // Realizando uma operação com os objetos
        Matriz resultadoSoma = matriz_A.somar(matriz_B);

        // Exibindo o resultado
        System.out.println(resultadoSoma);
    }
}
